src/coreclr/jit/morph.cpp

```diff
@@ -14534,6 +14534,38 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
     if (opts.OptimizationEnabled() && multiOp->OperIs(GT_HWINTRINSIC))
     {
         GenTreeHWIntrinsic* hw = multiOp->AsHWIntrinsic();
+
+        // Move constant vectors from op1 to op2 for commutative and compare operations
```

Comment:
It would be better to move this inside `fgOptimizeHWIntrinsic` (along with the XOR transform).